src/v/raft/consensus.cc

```diff
@@ -2214,7 +2219,7 @@ ss::future<install_snapshot_reply> consensus::finish_snapshot(
 
 ss::future<> consensus::write_snapshot(write_snapshot_cfg cfg) {
     model::offset last_included_index = cfg.last_included_index;
-    bool updated = co_await _op_lock
+    bool updated = co_await _snapshot_lock
```

Comment:
So browsing the usage sites of `_last_snapshot_index/term` I came across `recovery_stm` that uses those apparently without any lock... This is not directly related to the pr, but looks suspicious, right?

Reply:
> So browsing the usage sites of _last_snapshot_index/term I came across recovery_stm that uses those apparently without any lock... This is not directly related to the pr, but looks suspicious, right?

ping @mmaslankaprv ?

Reply:
oh, we had a chat with Alexey about that, it doesn't change the way `recovery_stm` access the snapshot as it is currently done without any lock